Tests/PackageGraphTests/DependencyResolverTests.swift

```diff
@@ -551,6 +551,70 @@ class DependencyResolverTests: XCTestCase {
        ])
     }
 
+    func testIncompleteMode() throws {
+        let provider = MockPackagesProvider(containers: [
+            MockPackageContainer(name: "A", dependencies: [
+                "1.0.0": [
+                    (container: "B", requirement: .versionSet(v1Range)),
+                ],
+            ]),
+            MockPackageContainer(name: "B", dependencies: [
+                "1.0.0": [],
+            ]),
+        ])
+
+        let resolver = MockDependencyResolver(provider, MockResolverDelegate())
+        resolver.incompleteMode = true
```

Comment:
Oh... would it be better to just make this a parameter and not use mutability, or make it an option on the resolve call?